### PR TITLE
Generic section plotting routine

### PR DESCRIPTION
**Pull request by jkrasting**
- Initial commit

### CODE REVIEW

File: om4labs/diags/generic_section_transport/generic_section_transport.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        return parser.parse_args(None).__dict__
+    else:
+        return parser.parse_args(cliargs)
+
+
+def read(dictArgs, varlist=["umo", "vmo"]):
+    """Reads umo and vmo files from a working directory"""
+
+    infile = dictArgs["infile"]
+
```
Comment (raphaeldussin) [inline om4labs/diags/generic_section_transport/generic_section_transport.py:46]: maybe idiotproof with:

```python

ds = xr.open_mfdataset(infile, use_cftime=True, combine="by_coords")
for var in varlist:
    if var not in ds.variables:
        raise ValueError(f"variable {var} not found in {dictArgs["infile"]}")

File: om4labs/diags/generic_section_transport/generic_section_transport.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    return transport
+
+
+def plot(transport, ylim=None, label=None, passage_label=None, obsrange=None):
+    fig = plt.figure(figsize=(8.5, 4))
+    ax = plt.subplot(1, 1, 1)
+
+    # plot obs range
+    if obsrange is not None:
+        minval = (transport.values * 0.0) + min(obsrange)
```
Comment (raphaeldussin) [inline om4labs/diags/generic_section_transport/generic_section_transport.py:143]: this is a bit easier to understand:

```python
minval = min(obsrange) * np.ones((len(transport.values)))

**Review by raphaeldussin — CHANGES_REQUESTED** [review]
a couple of nitpicks but freat PR

File: om4labs/diags/generic_section_transport/generic_section_transport.py (hunk 1 above)
Comment (jkrasting) [reply om4labs/diags/generic_section_transport/generic_section_transport.py]: Thanks, @raphaeldussin. I addressed this in f38464d.

File: om4labs/diags/generic_section_transport/generic_section_transport.py (hunk 2 above)
Comment (jkrasting) [reply om4labs/diags/generic_section_transport/generic_section_transport.py]: Thanks, @raphaeldussin. I addressed this in f38464d.

**Comment by raphaeldussin** [comment]
here are my comments after testing:
* it is pretty slow, which may be due to the glob calls
* it breaks the logic of applying to a list of files and assumes implicitly a certain directory structure by invoking it with the pp path

can we create individual transports by providing the list of files? say if I want to do only drake over 1950-2000 I will just pass the N corresponding files. And let DORA loop over the different sections?

**Comment by jkrasting** [comment]
> can we create individual transports by providing the list of files? say if I want to do only drake over 1950-2000 I will just pass the N corresponding files. And let DORA loop over the different sections?

This is what the generic section transport routine accomplishes and is consistent with that philosophy.  I view this routine here as more of a driver than a pure diagnostic routine.

**Comment by jkrasting** [comment]
@raphaeldussin - these changes should take care of everything we talked about this morning

**Pull request merged**